### ISSUE TITLE
OneofObject restriction on variant types

### ISSUE THREAD

**Issue by nicolaiunrein** [async-graphql/async-graphql#923]
Hey folks,
just played around with the OneofObject and wondered why there is a restriction on the inner type being unique? It is a tagged enum representation after all so imo this should be fine:
```rust
#[derive(OneofObject)]
enum Filter {
    Any(Vec<String>),
    All(Vec<String>)
}
```

but I get an error: `This type already used by another variant`

Note: there is actually a spelling mistake in the error message. It should say `This type is already used by another variant`

**Comment by sunli829** [comment]
It's a bug,will fix it now😃

**Comment by sunli829** [comment]
Fixed in `4.0.0-alpha.2`

**Comment by nicolaiunrein** [comment]
thanks
